File: sudoku-ums/ums-boot/src/main/java/com/zumin/sudoku/ums/mapper/SysResourceMapper.java
package com.zumin.sudoku.ums.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.zumin.sudoku.ums.pojo.entity.SysResource;
import java.util.List;
import org.apache.ibatis.annotations.Mapper;

@Mapper
public interface SysResourceMapper extends BaseMapper<SysResource> {

  List<SysResource> listWithRoles();
}

File: sudoku-ums/ums-boot/src/main/java/com/zumin/sudoku/ums/mapper/SysResourceMapper.kt
package com.zumin.sudoku.ums.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import com.zumin.sudoku.ums.pojo.SysResource
import org.apache.ibatis.annotations.Mapper

@Mapper
interface SysResourceMapper : BaseMapper<SysResource> {
  fun listWithRoles(): List<SysResource>
}